File: app/src/main/java/com/josegonzalez/musicsearchsample/view/itemDetails/ItemDetailActivity.java
package com.josegonzalez.musicsearchsample.view.itemDetails;

import android.os.Bundle;

import androidx.appcompat.widget.Toolbar;

import com.josegonzalez.musicsearchsample.R;
import com.josegonzalez.musicsearchsample.view.BaseActivity;

public class ItemDetailActivity extends BaseActivity {

    public static final String ARTIST_KEY = "artist";
    public static final String TRACK_KEY = "track";
    public static final String IMAGE_KEY = "image";
    public static final String ITEM_DETAIL_DEEP_LINK = HOST + "itemDetail";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_item_detail);
        Toolbar toolbar = findViewById(R.id.detail_toolbar);
        setSupportActionBar(toolbar);

        if (savedInstanceState == null) {
            Bundle arguments = new Bundle();
            arguments.putString(ARTIST_KEY, getIntent().getStringExtra(ARTIST_KEY));
            arguments.putString(TRACK_KEY, getIntent().getStringExtra(TRACK_KEY));
            arguments.putString(IMAGE_KEY, getIntent().getStringExtra(IMAGE_KEY));
            ItemDetailFragment fragment = new ItemDetailFragment();
            fragment.setArguments(arguments);
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.item_detail_container, fragment)
                    .commit();
        }
    }

}

File: app/src/main/java/com/josegonzalez/musicsearchsample/features/detail/ItemDetailActivity.kt
package com.josegonzalez.musicsearchsample.features.detail

import android.os.Bundle
import androidx.appcompat.widget.Toolbar
import com.josegonzalez.musicsearchsample.R
import com.josegonzalez.musicsearchsample.common.BaseActivity

class ItemDetailActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_item_detail)
        init(savedInstanceState)
    }

    private fun init(savedInstanceState: Bundle?) {
        val toolbar = findViewById<Toolbar>(R.id.detail_toolbar)
        setSupportActionBar(toolbar)

        if (savedInstanceState == null) {
            val arguments = Bundle()
            arguments.putString(ARTIST_KEY, intent.getStringExtra(ARTIST_KEY))
            arguments.putString(TRACK_KEY, intent.getStringExtra(TRACK_KEY))
            arguments.putString(IMAGE_KEY, intent.getStringExtra(IMAGE_KEY))

            val fragment = ItemDetailFragment()
            fragment.arguments = arguments
            supportFragmentManager.beginTransaction()
                    .add(R.id.item_detail_container, fragment)
                    .commit()
        }
    }

    companion object {
        const val ARTIST_KEY = "artist"
        const val TRACK_KEY = "track"
        const val IMAGE_KEY = "image"
    }

}